js/fetch.ts

```diff
@@ -20,15 +20,72 @@ import {
 } from "./fetch_types";
 import { TextDecoder } from "./text_encoding";
 
+type HeadersInit = Array<[string, string]> | { [K: string]: string };
+
+interface Header {
+  name: string;
+  value: string;
+}
```

Comment:
Seems superfluous?
I would think `this._append(name, value)` is readable and type-checked enough. (It also uses one less object.)

Reply:
It's used to store headers in `DenoHeader.headersList`, We can also use a tuple instead of an object, what about `[string, string]`

Reply:
I see, sorry - this is fine.